tests/commands/test_store_api.py

```diff
@@ -656,8 +656,8 @@ def test_upload_straightforward(client_mock, emitter, config):
     with patch.dict("charmcraft.commands.store.store.UPLOAD_ENDING_STATUSES", fake_statuses):
         result = store._upload(test_endpoint, test_filepath)
 
-    # check all client calls
-    assert client_mock.mock_calls == [
+    # check all client calls, excpet whoami() in first 10 calls
```

Comment:
why are there ten (!) `whoami()` calls? That's a lot!

Reply:
No, it just one, but since `whoami()` in the mock itself, it become:
```
E         +  call.whoami(),
E         +  call.whoami().__getitem__('account'),
E         +  call.whoami().__getitem__().__getitem__('display-name'),
E         +  call.whoami().__getitem__().__getitem__('username'),
E         +  call.whoami().__getitem__().__getitem__('id'),
E         +  call.whoami().__getitem__('packages'),
E         +  call.whoami().__getitem__('packages'),
E         +  call.whoami().__getitem__().__iter__(),
E         +  call.whoami().__getitem__('channels'),
E         +  call.whoami().__getitem__('permissions'),
```

Reply:
Oh I see. I think this shows that maybe we can improve the design a bit; the only thing we need is the `whoami()`  call on the client, right? All that other handling of packages and channels and permissions is unnecessary here. 
So what if we add a `_check_authorized()` method that just does `self._client_whoami()`? Then we can add the single `call.whoami()` call to these lists to document the fact that the upload methods call whoami.

What do you think?

Reply:
Hmmm, this also works.